### PR TITLE
Add listen bucket notification

### PR DESCRIPTION
**Pull request by dvaldivia**
One can subscribe to bucket notifications and get an `impl Stream<Item = NotificationInfo, Error = Err>`

```
c.listen_bucket_notificaion("cattest")
            .map_err(|_e| ())
            .for_each(|x| {
                println!("asdasd {:?}", x);
                Ok(())
            })
```

Also added license header.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "minio-rs"
 version = "0.1.0"
-authors = ["Aditya Manthramurthy <aditya.mmy@gmail.com>", "Krishnan Parthasarathi <krishnan.parthasarathi@gmail.com>"]
+authors = ["Aditya Manthramurthy <aditya.mmy@gmail.com>", "Krishnan Parthasarathi <krishnan.parthasarathi@gmail.com>", "Daniel Valdivia hi@danielvaldivia.com"]
```
Comment (donatello) [inline Cargo.toml]:
```suggestion
authors = ["MinIO Dev Team" <dev@min.io>]
```

File: src/minio.rs [hunk 2]
```diff
@@ -308,10 +337,63 @@ impl Client {
                 resp.into_body()
                     .concat2()
                     .map_err(|err| Err::HyperErr(err))
-                    .and_then(move |chunk| b2s(chunk.into_bytes()))
+                    .and_then(move |chunk| chunk_to_string(&chunk))
                     .and_then(|s| xml::parse_bucket_list(s))
             })
     }
+
+    /// get_bucket_location - Get location for the bucket_name.
```
Comment (donatello) [inline src/minio.rs]: Update comment.

File: src/minio.rs [hunk 3]
```diff
@@ -308,10 +337,63 @@ impl Client {
...
+            method: Method::GET,
+            bucket: Some(bucket_name.to_string()),
+            object: None,
+            headers: HeaderMap::new(),
+            query: query_params,
+            body: Body::empty(),
+            ts: time::now_utc(),
+        };
+        // We'll be streaming the events out of this
+        let (tx, rx) = mpsc::unbounded_channel();
```
Comment (donatello) [inline src/minio.rs]: Why do we need this approach? We can just return a stream from the response future directly (using maybe the flatten_stream() combinator). This approach is spawning an extra future.

File: src/minio.rs (hunk 2 above)
Comment (dvaldivia) [reply src/minio.rs]: With what? :S

File: src/minio.rs [hunk 4]
```diff
@@ -308,10 +335,67 @@ impl Client {
                 resp.into_body()
                     .concat2()
                     .map_err(|err| Err::HyperErr(err))
-                    .and_then(move |chunk| b2s(chunk.into_bytes()))
+                    .and_then(move |chunk| chunk_to_string(&chunk))
                     .and_then(|s| xml::parse_bucket_list(s))
             })
     }
+
+    /// listen_bucket_notificaion - Get bucket notifications for for the bucket_name.
```
Comment (donatello) [inline src/minio.rs]: s/for for/for/

File: src/minio.rs [hunk 5]
```diff
@@ -352,10 +381,67 @@ impl Client {
...
+            .map(|resp| {
+                // Read the whole body for bucket location response.
+                resp.into_body()
+                    .map_err(|e| Err::HyperErr(e))
+                    .filter(|c| {
+                        // filter out white spaces sent by the server to indicate it's still alive
+                        c[0] != SPACE_BYTE[0]
+                    })
+                    .map(|chunk| {
+                        // Split the chunk by lines and process
```
Comment (donatello) [inline src/minio.rs]:
```suggestion
                        // Split the chunk by lines and process.
                        // TODO: Handle case when partial lines are present in the chunk
```

**Pull request merged**